### ISSUE TITLE
[Old Log Form] Log date not changeable for passed events in old log form.

### ISSUE THREAD

**Issue by dennistreysa** [2Abendsegler/GClh#2418]
### Describe the bug

I like using the old style log-form since it keeps the last used date, which is nice if you want to log multiple caches in the past without changing the date every single time which is the case with the new form.

For passed events I can not click (and change) the event date for log-type "Attended". The calendar wont pop up.

### To Reproduce

1. Activate the old log-style in the settings ("Use old-fashioned log form to log non drafts related logs")
2. Set "Default event log type" to "Attended"
3. Go to a passed event you have not logged as attended (e.g. for me GC76543)
4. Select Log-Type "Attended"
5. Try to change the date via clicking on the date -> Calendar won't pop up
6. Changing to any other log-type will activate the calendar again.

After some testing, I realized, that this is not also a GClh helper problem, you can also come across this for events that you want to log as "Attended", but have not previously logged as "Will Attend", since GC will automatically add the URL parameter `logtype=10` in that case (which seems counter-intuitive to me and might as well be a mistake from Groundspeak).

Anyways, you can reproduce it via opening the old logpage with (GClh deactivated):

No logtype given, no problem since you have to change the logtype:
https://www.geocaching.com/seek/log.aspx?ID=6237205&lcn=1

Logtype set to "Attended", wrong log date
https://www.geocaching.com/seek/log.aspx?ID=6237205&lcn=1&logtype=10

### Expected behavior

Calendar should be active for passed events since the date is automatically set to "today" which makes no sense for events. Or it should be set to the event-date.

Workaround: Select any other logtype, then switch back to "Attended"->Date is correct now.

### OS

Windows

### Browser

Chrome

### GClh Version

0.14.6

### Additional context

I observed that a  `style="pointer-events: none;"` is set to the input field. If I manually remove that style via the DevTools, I can change the date again.

I think the problem boils down to some kind of race condition between the GC-Javascript and GClh.

If you open the logpage without having GClh activated, the default logtype is "- Select Type of Log -" (-1) and when you switch to "Attended" (10), the code in the "/controls/logbookpanel" script triggers and performs the following code:

```js
i == "10" ? ($("#uxDateVisited").datepicker("setDate", new Date($(selectors.EventDate).val())), $("#uxDateVisited").disabled = !0, $("#uxDateVisited").css("pointer-events", "none")) : ($("#uxDateVisited").disabled = !1, $("#uxDateVisited").css("pointer-events", ""))
```

Which essentially changes the log-date to the event-date and disables the datepicker if the logtype is "Attended".
This is also triggered when you switch to another logtype and switch back to "Attended".

Upon further investigating, I found out, that this behaviour can also manually be triggered via dispatching a change-event on the logtype-element via:

```js
var event = new Event('change');
var element = document.getElementById("ctl00_ContentBody_LogBookPanel1_ddLogType");
element.dispatchEvent(event);
```

Or in short, by using jQuery:

```js
$("#ctl00_ContentBody_LogBookPanel1_ddLogType").trigger("change");
```

Unfortunately, it seems that one can not simply add this code to GClh (somewhere after [this line](https://github.com/2Abendsegler/GClh/blob/9ae70bf7aa8c5390df6989a1b82c4eb3911da3f6/gc_little_helper_II.user.js#L4469) ), since the GC-Script seems to be not yet loaded at this point.

The solution I came up with (which will also solve the issue when `logtype=10` is set) is as follow:

Add the following lines after [this line](https://github.com/2Abendsegler/GClh/blob/9ae70bf7aa8c5390df6989a1b82c4eb3911da3f6/gc_little_helper_II.user.js#L4475C18-L4475C18):

```js
// Set DateVisited to EventDate when selected logtype is "Attended"
if (select.value == "10")
{
    var logdate = $("#uxDateVisited");
    var logdate_format = logdate.attr("userdateformat");
    var logdate_date = new Date($("#ctl00_ContentBody_LogBookPanel1_EventDate").val());
    var logdate_text = $.datepicker.formatDate(logdate_format, logdate_date);

    logdate.attr("value", logdate_text);
}
```

**Comment by 2Abendsegler** [comment]
Thank you for reporting the bug in detail and thank you also for the suggested solutions. I can reproduce the bug. We will fix the bug in the next version. I should have some time again at the beginning of October.

**Comment by capoaira** [comment]
merged with #2431
